Orange/widgets/visualize/owmosaic.py

```diff
@@ -676,80 +637,48 @@ def draw_text(side, attr, x0_x1, y0_y1,
 
             # calculate position of first attribute
             currpos = 0
-
             total, counts = get_counts(attr_vals, values)
-
             aligns = [Qt.AlignTop | Qt.AlignHCenter,
                       Qt.AlignRight | Qt.AlignVCenter,
                       Qt.AlignBottom | Qt.AlignHCenter,
                       Qt.AlignLeft | Qt.AlignVCenter]
             align = aligns[side]
             for i, val in enumerate(values):
-                perc = counts[i] / float(total)
                 if distributiondict[val] != 0:
+                    perc = counts[i] / float(total)
+                    xs = [x0 + currpos + width * 0.5 * perc,
+                          x0 - self.ATTR_VAL_OFFSET,
+                          x0 + currpos + width * perc * 0.5,
+                          x1 + self.ATTR_VAL_OFFSET]
+                    ys = [y1 + self.ATTR_VAL_OFFSET,
+                          y0 + currpos + height * 0.5 * perc,
+                          y0 - self.ATTR_VAL_OFFSET,
+                          y0 + currpos + height * 0.5 * perc]
                     if side == 0:
```

Comment:
Feature values are now shown only for the first feature (bottom side). Is this intentional? If it is, why setting xs and ys for other sides as well?